plugins/out_s3/arrow/compress.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * This converts S3 plugin's request buffer into Apache Arrow format.
+ *
+ * We use GLib binding to call Arrow functions (which is implemented
+ * in C++) from Fluent Bit.
+ *
+ * https://github.com/apache/arrow/tree/master/c_glib
+ */
+
+#include <arrow-glib/arrow-glib.h>
+#include <inttypes.h>
+
+/*
+ * GArrowTable is the central structure that represents "table" (a.k.a.
+ * matrix).
```

Comment:
I think that "data frame" is better than "matrix".
It seems that "matrix" uses the same type for all elements but "data frame" may use different type for each column.

Reply:
Fixed.